src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -1299,6 +1299,58 @@ public async Task TestConfigIsValid()
             }
         }
 
+        /// <summary>
+        /// Test to verify that provided value of depth-limit in the config file should be greater than 0.
+        /// -1 and null are special values to disable depth limit.
+        /// This test validates that depth-limit outside the valid range should fail validation.
+        /// </summary>
+        [DataTestMethod]
+        [DataRow(-1, DisplayName = "[PASS]: Valid Value: -1 to disable depth limit")]
+        [DataRow(0, DisplayName = "[FAIL]: Invalid Value: 0 for depth-limit.")]
+        [DataRow(2, DisplayName = "[PASS]: Valid Value: 2 for depth-limit.")]
+        [DataRow(null, DisplayName = "[PASS]: Valid Value: null for depth-limit.")]
+        [TestCategory(TestCategory.MSSQL)]
+        public async Task TestValidateConfigForDifferentDepthLimit(int? depthLimit)
+        {
+            // Fetch the MS_SQL integration test config file.
+            TestHelper.SetupDatabaseEnvironment(MSSQL_ENVIRONMENT);
+            FileSystemRuntimeConfigLoader testConfigPath = TestHelper.GetRuntimeConfigLoader();
+            RuntimeConfig configuration = TestHelper.GetRuntimeConfigProvider(testConfigPath).GetConfig();
+            configuration = configuration with
+            {
+                Runtime = configuration.Runtime with
+                {
+                    GraphQL = configuration.Runtime.GraphQL with { DepthLimit = depthLimit, UserProvidedDepthLimit = true }
+                }
+            };
+            const string CUSTOM_CONFIG = "custom-config.json";
+
+            MockFileSystem fileSystem = new();
+
+            // write it to the custom-config file and add it to the filesystem.
+            fileSystem.AddFile(CUSTOM_CONFIG, new MockFileData(configuration.ToJson()));
+            FileSystemRuntimeConfigLoader configLoader = new(fileSystem);
+            configLoader.UpdateConfigFilePath(CUSTOM_CONFIG);
+            RuntimeConfigProvider configProvider = TestHelper.GetRuntimeConfigProvider(configLoader);
+
+            Mock<ILogger<RuntimeConfigValidator>> configValidatorLogger = new();
+            RuntimeConfigValidator configValidator =
+                new(
+                    configProvider,
+                    fileSystem,
+                    configValidatorLogger.Object,
+                    true);
+
+            if (depthLimit is not null && depthLimit <= 0 && depthLimit != -1)
+            {
+                Assert.IsFalse(await configValidator.TryValidateConfig(CUSTOM_CONFIG, TestHelper.ProvisionLoggerFactory()));
+            }
+            else
+            {
+                Assert.IsTrue(await configValidator.TryValidateConfig(CUSTOM_CONFIG, TestHelper.ProvisionLoggerFactory()));
+            }
```

Comment:
can we avoid duplicating dab engine logic inside the test?
```csharp
if (depthLimit is not null && depthLimit <= 0 && depthLimit != -1)
```

This logic may change and leaks implementation into the test. 

I'd argue that you can improve this with data row input params:
`(int userProvidedDepthLimit, bool isValidInput)`
and then use Assert.AreEqual

also the {arrange, act, assert} can have 
```csharp
// Act
bool isConfigValidationSuccessful = await configValidator.TryValidateConfig(CUSTOM_CONFIG, TestHelper.ProvisionLoggerFactory());

// Assert
// <asserts>
```

Reply:
thanks good point